lib/ansible/modules/network/avi/avi_cloud.py

```diff
@@ -0,0 +1,243 @@
+            - Ipam profile for the cloud.
+            - It is a reference to an object of type ipamdnsproviderprofile.
+    license_type:
+        description:
+            - If no license type is specified then default license enforcement for the cloud type is chosen.
+            - The default mappings are container cloud is max ses, openstack and vmware is cores and linux it is sockets.
+            - Enum options - LIC_BACKEND_SERVERS, LIC_SOCKETS, LIC_CORES, LIC_HOSTS.
+    linuxserver_configuration:
+        description:
+            - Linuxserverconfiguration settings for cloud.
+    mesos_configuration:
+        description:
+            - Mesosconfiguration settings for cloud.
+    mtu:
+        description:
+            - Mtu setting for the cloud.
+            - Default value when not specified in API or module is interpreted by Avi Controller as 1500.
+    name:
+        description:
+            - Name of the object.
+        required: true
+    nsx_configuration:
+        description:
+            - Configuration parameters for nsx manager.
+            - Field introduced in 17.1.1.
+    obj_name_prefix:
+        description:
+            - Default prefix for all automatically created objects in this cloud.
+            - This prefix can be overridden by the se-group template.
+    openstack_configuration:
+        description:
+            - Openstackconfiguration settings for cloud.
+    oshiftk8s_configuration:
+        description:
+            - Oshiftk8sconfiguration settings for cloud.
+    prefer_static_routes:
+        description:
+            - Prefer static routes over interface routes during virtualservice placement.
+            - Default value when not specified in API or module is interpreted by Avi Controller as False.
+    proxy_configuration:
+        description:
+            - Proxyconfiguration settings for cloud.
+    rancher_configuration:
+        description:
+            - Rancherconfiguration settings for cloud.
+    tenant_ref:
+        description:
+            - It is a reference to an object of type tenant.
+    url:
+        description:
+            - Avi controller URL of the object.
+    uuid:
+        description:
+            - Unique object identifier of the object.
+    vca_configuration:
+        description:
+            - Vcloudairconfiguration settings for cloud.
+    vcenter_configuration:
+        description:
+            - Vcenterconfiguration settings for cloud.
+    vtype:
+        description:
+            - Cloud type.
+            - Enum options - CLOUD_NONE, CLOUD_VCENTER, CLOUD_OPENSTACK, CLOUD_AWS, CLOUD_VCA, CLOUD_APIC, CLOUD_MESOS, CLOUD_LINUXSERVER, CLOUD_DOCKER_UCP,
+            - CLOUD_RANCHER, CLOUD_OSHIFT_K8S.
+            - Default value when not specified in API or module is interpreted by Avi Controller as CLOUD_NONE.
+        required: true
+extends_documentation_fragment:
+    - avi
+'''
+
+
+EXAMPLES = '''
+  - name: Create a VMWare cloud with write access mode
```

Comment:
I'm happy for you to raise one PR that fixes that across any/all modules it affects at a later point.